services/storage/src/simcore_service_storage/api/v0/openapi.yaml

```diff
@@ -612,61 +620,79 @@ paths:
             application/json:
               schema:
                 $ref: '#/components/schemas/Envelope_TaskGet_'
-  '/v0/simcore-s3/folders/{folder_id}':
+  /v0/simcore-s3/folders/{folder_id}:
     delete:
       tags:
-        - simcore-s3
+      - simcore-s3
       summary: delete folders from project
       description: removes folders from a project
       operationId: delete_folders_of_project
       parameters:
-        - required: true
-          schema:
-            title: Folder Id
-            type: string
-          name: folder_id
-          in: path
-        - required: true
-          schema:
-            title: User Id
-            exclusiveMinimum: true
-            type: integer
-            minimum: 0
-          name: user_id
-          in: query
-        - required: false
-          schema:
-            title: Node Id
-            type: string
-            format: uuid
-          name: node_id
-          in: query
+      - required: true
+        schema:
+          type: string
+          title: Folder Id
+        name: folder_id
+        in: path
+      - required: true
+        schema:
+          type: integer
+          exclusiveMinimum: true
+          title: User Id
+          minimum: 0
+        name: user_id
+        in: query
+      - required: false
+        schema:
+          type: string
+          format: uuid
+          title: Node Id
+        name: node_id
+        in: query
       responses:
         '204':
           description: Successful Response
-  '/v0/simcore-s3/files/metadata:search':
+  /v0/simcore-s3/files/metadata:search:
     post:
       tags:
-        - simcore-s3
+      - simcore-s3
       summary: search for files starting with
-      description: search for files starting with `startswith` in the file_meta_data table
-      operationId: search_files_starting_with
+      description: search for files starting with `startswith` and/or matching a sha256_checksum
+        in the file_meta_data table
+      operationId: search_files
       parameters:
-        - required: true
-          schema:
-            title: User Id
-            exclusiveMinimum: true
-            type: integer
-            minimum: 0
-          name: user_id
-          in: query
-        - required: false
-          schema:
-            title: Startswith
-            type: string
-            default: ''
-          name: startswith
-          in: query
+      - required: true
+        schema:
+          type: integer
+          exclusiveMinimum: true
+          title: User Id
+          minimum: 0
+        name: user_id
+        in: query
+      - required: false
+        schema:
+          type: string
+          title: Startswith
+          default: ''
+        name: startswith
+        in: query
+      - required: false
+        schema:
+          type: string
+          pattern: ^[a-fA-F0-9]{64}$
+          title: Sha256 Checksum
+        name: sha256_checksum
+        in: query
+      - required: false
```

Comment:
Ideally it should, but it has not yet been implemented in the frontend. So when a file is uploaded there, no checksum is passed along... Hence, for now it will not be required.

Reply:
I finally understand the reasoning behind this one.